nuitka/tools/specialize/templates_python/StrNodes.py.j2

```diff
@@ -0,0 +1,137 @@
+#     Copyright 2022, Fire-Cube <ben7@gmx.ch>
+#
+#     Part of "Nuitka", an optimizing Python compiler that is compatible and
+#     integrates with CPython, but also works on its own.
+#
+#     Licensed under the Apache License, Version 2.0 (the "License");
+#     you may not use this file except in compliance with the License.
+#     You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#     Unless required by applicable law or agreed to in writing, software
+#     distributed under the License is distributed on an "AS IS" BASIS,
+#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#     See the License for the specific language governing permissions and
+#     limitations under the License.
+#
+""" Specialized str nodes
+
+WARNING, this code is GENERATED. Modify the template StrNodes.py.j2 instead!
+"""
+
+from abc import abstractmethod
+
+from .ConstantRefNodes import makeConstantRefNode
+from .ExpressionBases import (
+    ExpressionChildHavingBase,
+    ExpressionChildrenHavingBase,
+)
+from .ExpressionShapeMixins import (
+    ExpressionBoolShapeExactMixin,
+    ExpressionBytesShapeExactMixin,
+    ExpressionIntShapeExactMixin,
+    ExpressionListShapeExactMixin,
+    ExpressionStrOrUnicodeExactMixin,
+    ExpressionStrShapeExactMixin,
+    ExpressionTupleShapeExactMixin,
+)
+from .NodeBases import SideEffectsFromChildrenMixin
+from .NodeMetaClasses import NodeCheckMetaClass
+
+
+def getStrOperationClasses():
+    """Return all str operation nodes, for use by code generation."""
+    return (
+        cls
+        for kind, cls in NodeCheckMetaClass.kinds.items()
+        if kind.startswith("EXPRESSION_STR_OPERATION_")
+    )
+
+{% for operation_name, arg_name, shape, is_base, inheritance_base, base_name, simulator, simulator_return_value in specifications %}
+class ExpressionStrOperation{{ operation_name }}(
+    {% if inheritance_base %}
+    ExpressionStrOperation{{ base_name }}Base
```

Comment:
Actually I would like to get rid of the base classes when we do the generation, i.e. their content is generated into the class.

Reply:
Cool, that makes a lot of things easier.